osu.Game/Screens/Edit/Setup/ResourcesSection.cs

```diff
@@ -40,93 +39,110 @@ private void load()
         {
             Children = new Drawable[]
             {
-                backgroundTextBox = new FileChooserLabelledTextBox(".jpg", ".jpeg", ".png")
+                backgroundChooser = new LabelledFileChooser(".jpg", ".jpeg", ".png")
                 {
                     Label = "Background",
                     FixedLabelWidth = LABEL_WIDTH,
-                    PlaceholderText = "Click to select a background image",
-                    Current = { Value = working.Value.Metadata.BackgroundFile },
                     TabbableContentContainer = this
                 },
-                audioTrackTextBox = new FileChooserLabelledTextBox(".mp3", ".ogg")
+                audioTrackChooser = new LabelledFileChooser(".mp3", ".ogg")
                 {
                     Label = "Audio Track",
                     FixedLabelWidth = LABEL_WIDTH,
-                    PlaceholderText = "Click to select a track",
-                    Current = { Value = working.Value.Metadata.AudioFile },
                     TabbableContentContainer = this
                 },
             };
 
-            backgroundTextBox.Current.BindValueChanged(backgroundChanged);
-            audioTrackTextBox.Current.BindValueChanged(audioTrackChanged);
+            if (!string.IsNullOrEmpty(working.Value.Metadata.BackgroundFile))
+                backgroundChooser.Current.Value = new FileInfo(working.Value.Metadata.BackgroundFile);
+
+            if (!string.IsNullOrEmpty(working.Value.Metadata.AudioFile))
+                audioTrackChooser.Current.Value = new FileInfo(working.Value.Metadata.AudioFile);
+
+            backgroundChooser.Current.BindValueChanged(backgroundChanged, true);
+            audioTrackChooser.Current.BindValueChanged(audioTrackChanged, true);
```

Comment:
These callbacks being `BindValueChanged(..., true)` means that the filenames will be normalised potentially without the user's explicit intervention. I don't think this is a good idea. Fine to do it on save or on change, but existing maps shouldn't be magically changed IMO.

I haven't tested but I'm like 90% sure you'll be able to break maps with this too. I'd try entering editor, going to metadata tab, then exiting editor without saving - I'm pretty sure the background will be moved in the file store but `.osu` file won't be updated, and so if you restart the game (to bust working beatmap cache), the map background will be gone.